### PR TITLE
[Php80] Skip no result usage on NullsafeOperatorRector (#6680)

### PR DESCRIPTION
**Pull request by echo511**
Fixes https://github.com/rectorphp/rector/issues/6680

### CODE REVIEW

**Comment by samsonasik** [comment]
@TomasVotruba new symplify release seems cause error:

```bash
bin/rector process config utils scoper.php --ansi --no-progress-bar
  shell: /usr/bin/bash -e {0}
  env:
    COMPOSER_ROOT_VERSION: dev-main

PHP Fatal error:  Uncaught _PHPStan_0ebfea013\Nette\FileNotFoundException: File '/home/runner/work/rector-src/rector-src/vendor/symplify/phpstan-rules/packages/object-calisthenics/config/object-calisthenics-services.neon' is missing or is not readable. in phar:///home/runner/work/rector-src/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/nette/di/src/DI/Config/Loader.php:29
```

**Comment by samsonasik** [comment]
I create PR for PHPStan includes not exists neon file https://github.com/rectorphp/rector-src/pull/828

**Comment by samsonasik** [comment]
@echo511 please rebase

**Comment by echo511** [comment]
@samsonasik done

**Comment by samsonasik** [comment]
Please run the following commands:

```bash
# rectify
bin/rector process rules/Php80/Rector/If_/NullsafeOperatorRector.php

# fix cs 
composer fix-cs -- rules/Php80/Rector/If_/NullsafeOperatorRector.php

# rectify again to ensure all ok
bin/rector process rules/Php80/Rector/If_/NullsafeOperatorRector.php
```

File: rules/Php80/Rector/If_/NullsafeOperatorRector.php [hunk 1]
```diff
@@ -246,6 +248,15 @@ private function shouldProcessAssignInCurrentNode(Assign $assign, Node $nextNode
             return false;
         }
 
+        if (
+            property_exists($assign->expr, self::VAR) &&
+            property_exists($nextNode->expr, self::VAR) &&
```
Comment (TomasVotruba) [inline rules/Php80/Rector/If_/NullsafeOperatorRector.php:253]: This check is quite magical. PHPStan and Rector would not understand it and fail to refactor in the future.
It should be handled instanceof with specific classes.

Comment (echo511) [reply rules/Php80/Rector/If_/NullsafeOperatorRector.php:253]: Well, I took inspiration from the code above. If it's a problem perhaps wider refactoring of the class would proof beneficial. @TomasVotruba

Comment (samsonasik) [reply rules/Php80/Rector/If_/NullsafeOperatorRector.php:253]: You may can change the code something like this:

```php
    private function shouldProcessAssignInCurrentNode(Assign $assign, Node $nextNode): bool
    {
        if (! $nextNode instanceof Return_ && ! $nextNode instanceof Expression) {
            return false;
        }

        if ($nextNode->expr instanceof Assign) {
            return false;
        }

        if (! $assign->expr instanceof MethodCall) {
            return ! $this->valueResolver->isNull($nextNode->expr);
        }

        if (! $nextNode->expr instanceof MethodCall) {
            return ! $this->valueResolver->isNull($nextNode->expr);
        }

        if (! $this->nodeComparator->areNodesEqual($assign->expr->var, $nextNode->expr->var)) {
            return ! $this->valueResolver->isNull($nextNode->expr);
        }

        return false;
    }
```

after than, run the following commands:

```bash
# rectify
bin/rector process rules/Php80/Rector/If_/NullsafeOperatorRector.php

# fix cs 
composer fix-cs -- rules/Php80/Rector/If_/NullsafeOperatorRector.php

# rectify again to ensure all ok
bin/rector process rules/Php80/Rector/If_/NullsafeOperatorRector.php
```

**Comment by samsonasik** [comment]
@echo511 I cherry-picked your commit with check instanceof usage at https://github.com/rectorphp/rector-src/pull/838

**Pull request closed without merging**